securesystemslib/dsse.py

```diff
@@ -26,7 +26,7 @@ def __init__(
     ):
```

Comment:
I wasn't sure if I should change the last param from signatures: List[Signature] to signatures: Dict[str, Signature], so any information is appreciated

Reply:
Good question! Please do change the constructor argument to `Dict[str, Signature]` and implement the translation in `from_dict` and `to_dict`. This is also what we do with [python-tuf  `Metadata`](https://github.com/theupdateframework/python-tuf/blob/e1b52e7fd828c11159a12aef4259133c3f05accf/tuf/api/metadata.py#L125).

Reply:
I also just saw that we raise, if there are [duplicate (by keyid)](https://github.com/theupdateframework/python-tuf/blob/e1b52e7fd828c11159a12aef4259133c3f05accf/tuf/api/metadata.py#L189-L195) signatures there. I suggest to do the same in `Envelope.from_dict`.